### PR TITLE
Use generator in Mailbox.get_mail to limit memory leak

### PR DESCRIPTION
**Pull request by ad-m**
Hello,

I see no reason why messages should be collected on the list. This causes huge memory usage. I am importing 15 GB mbox files. You should delete messages as soon as possible (references to them) to reduce memory usage.

### CODE REVIEW

**Comment by ad-m** [comment]
@coddingtonbear , it's hard to provide multiple slots (accounts) for integration tests? We can use hashing of some environment variables eg. ``TRAVIS_JOB_ID`` (https://docs.travis-ci.com/user/environment-variables/#Convenience-Variables) to select account. I mean eg. ``job_id%slot_count``. Then we reduce risk of collision.

**Comment by coddingtonbear** [comment]
Probably not difficult, no; until you began posting PRs, mostly it was just me, so collisions were unlikely enough that I've never invested the time in preventing them.  Since they _do_ happen a little more often now, I'll try to find a little time to set up more accounts (or alter the tests to be more forgiving of multiple simultaneous integration tests).

File: django_mailbox/tests/base.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def _get_new_messages(self, mailbox, condition=None):
                 raise EmailIntegrationTimeout()
             messages = self.mailbox.get_new_mail(condition)
             if messages:
```
Comment (ivlevdenis) [inline django_mailbox/tests/base.py]:
```python
if messages:
```
All time is True, messages is generator!

Comment (coddingtonbear) [inline django_mailbox/tests/base.py]: Ooo -- good catch, @ivlevdenis!

File: django_mailbox/tests/base.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def _get_new_messages(self, mailbox, condition=None):
...
-                return messages
+                return list(messages)
```
Comment (ivlevdenis) [inline django_mailbox/tests/base.py]: It is strange to use a generator if it still converts to a list.

**Pull request merged**